src/Monocle/Api/Test.hs

```diff
@@ -26,20 +38,40 @@ mkAppEnv workspace = withLogger \glLogger -> do
       aEnv = Env {..}
   pure $ AppEnv {..}
 
-withTestApi :: IO AppEnv -> (Logger -> MonocleClient -> Assertion) -> IO ()
+--  Note: when running Effect, the order is set
+runAppEnv :: AppEnv -> Eff (ElasticEffect : MonoConfigEffect : E.Reader AppEnv : LoggerEffect : E.Error ServerError : E.Fail : E.Concurrent : IOE : '[]) a -> IO a
```

Comment:
```suggestion
runAppEnv :: AppEnv -> Eff ('[ElasticEffect, MonoConfigEffect, E.Reader AppEnv, LoggerEffect, E.Error ServerError, E.Fail, E.Concurrent, IOE]) a -> IO a
```

Reply:
I don't quite get why sometime the Effects needs to be listed in order, and sometime a constraint is enough, hence the weird signature here and there. Thank you for the suggestion!

Reply:
When you use the effects, order does not matter (as long as the effects are listed in the constraints).

However, whenever you interpret them, you have to "prove" that all the effets are interpreted and to do so, you need a concrete type.

I don't know for `effectful`, but for `polysemy`, you can put an interpreter (or an interceptor) anywhere in the code, and all the effects will be "removed" during your "final" interpreter run, without breaking your code.
It works because your effect library consider that, your type not being fully define, an "additional" (duplicated) effect was present, and so, it removes it.

Reply:
I see thank you very much for confirming. I guess the order is necessary anywhere Eff is on the left side of the arrow and IO is on the right side. For example, this `withTenant` doesn't interpret the effect locally, but it's like contaminated by the call to `withTenantConfig` below:

https://github.com/change-metrics/monocle/blob/0b806b2beaacf47d0dff03f909b2a179320d8f78/src/Monocle/Backend/Test.hs#L88-L111

Reply:
> I guess the order is necessary anywhere Eff is on the left side of the arrow and IO is on the right side

Usually yes, but it can change with hofs:

```queryAsync :: (ElasticEffect :> es, MonadIO m) => (Eff es a :> m a) -> Eff es a -> Eff es a```